### PR TITLE
TSQL - Implement unary and binary expressions with correct precedence

### PR DESCRIPTION
**Pull request by jimidle**
Unary operators were not correctly implemented in the ANTLR grammar file, and this is now corrected. In addition, unary operators are right associative, which affects order of operations in certain case. Precedence has also been corrected to follow the TSQL normative specification. 

Progresses: #275

### CODE REVIEW

**Comment by jimidle** [comment]
@vil1 @nfx I have another PR based on this one that I would like to submit if eiether of you have time to check and merge this.

File: core/pom.xml
Comment (vil1) [inline core/pom.xml:93]:
```suggestion
      <version>${mockito.version}</version>
```

File: core/src/main/antlr4/com/databricks/labs/remorph/parsers/tsql/TSqlLexer.g4
Comment (vil1) [inline core/src/main/antlr4/com/databricks/labs/remorph/parsers/tsql/TSqlLexer.g4:1215]:
```suggestion
//TODO: In ANTLR v4 we cannot distinguish between 7 + ---42 and  -- A comment. Needs custom char stream
```

File: core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala
Comment (vil1) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala:74]:
```suggestion
// Binary bitwise expressions
```

**Review by vil1 — APPROVED** [review]
LGTM (modulo a few minute, non-blocking suggestions)

File: core/pom.xml
Comment (jimidle) [reply core/pom.xml:93]: Mockito was added by someone else. It needs to be in the Pom or IntelliJ will not run any tests because snowflake uses it.

**Comment by jimidle** [comment]
Please merge so I can rebase the other outstanding PRs :)

**Comment by jimidle** [comment]
> LGTM (modulo a few minute, non-blocking suggestions)

@vil1 Do you have merge permissions on this repo? I am not completely blocked, but the other PRs will need to remain WIP until this one is merged. 

#307-> #308  -> #309

**Comment by vil1** [comment]
Sorry but I don't have the required permissions

**Comment by sundarshankar89** [comment]
@nfx  Need your attention codecov has been waiting on status for last 6+ hours.

**Review by sundarshankar89 — APPROVED** [review]
LGTM

**Pull request merged**
